src/collectives_c.c

```diff
@@ -392,9 +392,14 @@ shmem_longdouble_min_to_all(long double *target, long double *source, int nreduc
 {
     SHMEM_ERR_CHECK_INITIALIZED();
 
-    shmem_internal_op_to_all(target, source, nreduce, sizeof(long double),
+    if(!shmem_long_dub_supported) {
```

Comment:
Push this conditional into shmem_internal_op_to_all instead of in the language bindings.  We anticipate needing a SW solution for float/double complex and potentially other type/op combinations.